tools/test.py

```diff
@@ -395,16 +395,8 @@ def HasRun(self, output):
 
 
     duration = output.test.duration
-
-    # total_seconds() was added in 2.7
-    total_seconds = (duration.microseconds +
-      (duration.seconds + duration.days * 24 * 3600) * 10**6) / 10**6
-
-    # duration_ms is measured in seconds and is read as such by TAP parsers.
-    # It should read as "duration including ms" rather than "duration in ms"
     logger.info('  ---')
-    logger.info('  duration_ms: %d.%d' %
-      (total_seconds, duration.microseconds / 1000))
+    logger.info('  duration_ms: %.5f' % (duration  / timedelta(milliseconds=1)))
```

Comment:
Are you sure that works? When I try it in my python REPL, I'm getting an error:

```console
> '  duration_ms: %.5f' %  timedelta(milliseconds=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: must be real number, not datetime.timedelta
```

Reply:
I think there might need` .total_seconds()` to convert the class to a number

Reply:
I tested this running `tools/test.py test/parallel/test-runner-option-validation.js` (or any other file)

Reply:
> I tested this running `tools/test.py test/parallel/test-runner-option-validation.js` (or any other file)

I don't think your example script runs that line, you should add ` --progress tap` on the suffix

Reply:
I think it runs well. because `duration / timedelta()` is a float instead of a class

Reply:
you should add `--progress tap` on the suffix

Reply:
> Are you sure that works? When I try it in my python REPL, I'm getting an error:
> 
> ```
> > '  duration_ms: %.5f' %  timedelta(milliseconds=1)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> TypeError: must be real number, not datetime.timedelta
> ```

i runs well because `duration  / timedelta(milliseconds=1)` is a number.

Reply:
% `python3`
```
>>> from datetime import timedelta
>>> duration = 5
>>> '  duration_ms: %.5f' % (duration  / timedelta(milliseconds=1))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for /: 'int' and 'datetime.timedelta'
```

Reply:
`duration` is not a number though, I think it's supposed to be a `timedelta` object.